Lib9c/Action/SellCancellation.cs

```diff
@@ -78,72 +85,82 @@ public override IAccountStateDelta Execute(IActionContext context)
             var started = DateTimeOffset.UtcNow;
             Log.Verbose("{AddressesHex}Sell Cancel exec started", addressesHex);
 
-            if (!states.TryGetAgentAvatarStates(ctx.Signer, sellerAvatarAddress, out _, out var avatarState))
+            if (!states.TryGetAvatarState(ctx.Signer, sellerAvatarAddress, out var avatarState))
             {
-                return states;
+                throw new FailedLoadStateException(
+                    $"{addressesHex}Aborted as the avatar state of the seller was failed to load.");
             }
             sw.Stop();
             Log.Verbose("{AddressesHex}Sell Cancel Get AgentAvatarStates: {Elapsed}", addressesHex, sw.Elapsed);
             sw.Restart();
 
-            if (!avatarState.worldInformation.TryGetUnlockedWorldByStageClearedBlockIndex(
-                out var world))
-                return states;
-
-            if (world.StageClearedId < GameConfig.RequireClearedStageLevel.ActionsInShop)
+            if (!avatarState.worldInformation.IsStageCleared(GameConfig.RequireClearedStageLevel.ActionsInShop))
             {
-                // 스테이지 클리어 부족 에러.
-                return states;
+                avatarState.worldInformation.TryGetLastClearedStageId(out var current);
+                throw new NotEnoughClearedStageLevelException(addressesHex,
+                    GameConfig.RequireClearedStageLevel.ActionsInShop, current);
             }
 
-            if (!states.TryGetState(ShopState.Address, out Bencodex.Types.Dictionary shopStateDict))
+            if (!states.TryGetState(shardedShopAddress, out Dictionary shopStateDict))
             {
-                return states;
+                ShardedShopState shopState = new ShardedShopState(shardedShopAddress);
+                shopStateDict = (Dictionary) shopState.Serialize();
             }
             sw.Stop();
             Log.Verbose("{AddressesHex}Sell Cancel Get ShopState: {Elapsed}", addressesHex, sw.Elapsed);
             sw.Restart();
 
             // 상점에서 아이템을 빼온다.
-            Dictionary products = (Dictionary)shopStateDict["products"];
+            List products = (List)shopStateDict[ProductsKey];
+
+            IValue productIdSerialized = productId.Serialize();
+            Dictionary productSerialized = products
+                .Select(p => (Dictionary) p)
+                .FirstOrDefault(p => p[ProductIdKey].Equals(productIdSerialized));
 
-            IKey productIdSerialized = (IKey)productId.Serialize();
-            if (!products.ContainsKey(productIdSerialized))
+            bool backWardCompatible = false;
```

Comment:
```suggestion
            bool backwardCompatible = false;
```

[*Backward*](https://dictionary.cambridge.org/dictionary/english/backward) seems one word, so is it better to use the part as lowercase? 🤔